[2025-10-05T08:30:47.497+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-10-05T08:30:47.510+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: forest_cover_mlops_pipeline.train_and_log_models scheduled__2025-10-05T08:25:00+00:00 [queued]>
[2025-10-05T08:30:47.514+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: forest_cover_mlops_pipeline.train_and_log_models scheduled__2025-10-05T08:25:00+00:00 [queued]>
[2025-10-05T08:30:47.515+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 3
[2025-10-05T08:30:47.521+0000] {taskinstance.py:2888} INFO - Executing <Task(_PythonDecoratedOperator): train_and_log_models> on 2025-10-05 08:25:00+00:00
[2025-10-05T08:30:47.528+0000] {standard_task_runner.py:72} INFO - Started process 3981 to run task
[2025-10-05T08:30:47.530+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'forest_cover_mlops_pipeline', 'train_and_log_models', 'scheduled__2025-10-05T08:25:00+00:00', '--job-id', '88', '--raw', '--subdir', 'DAGS_FOLDER/forest_cover_dag.py', '--cfg-path', '/home/airflow/tmp/tmp59use1t_']
[2025-10-05T08:30:47.531+0000] {standard_task_runner.py:105} INFO - Job 88: Subtask train_and_log_models
[2025-10-05T08:30:47.571+0000] {task_command.py:467} INFO - Running <TaskInstance: forest_cover_mlops_pipeline.train_and_log_models scheduled__2025-10-05T08:25:00+00:00 [running]> on host 3702e9b7dcde
[2025-10-05T08:30:47.925+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='mlops-team' AIRFLOW_CTX_DAG_ID='forest_cover_mlops_pipeline' AIRFLOW_CTX_TASK_ID='train_and_log_models' AIRFLOW_CTX_EXECUTION_DATE='2025-10-05T08:25:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-05T08:25:00+00:00'
[2025-10-05T08:30:47.925+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-10-05T08:30:48.083+0000] {logging_mixin.py:190} INFO - Entrenando 3 modelos con GridSearchCV...
[2025-10-05T08:30:48.084+0000] {logging_mixin.py:190} INFO - 
==================================================
[2025-10-05T08:30:48.084+0000] {logging_mixin.py:190} INFO - Modelo: RandomForest
[2025-10-05T08:30:48.084+0000] {logging_mixin.py:190} INFO - ==================================================
[2025-10-05T08:30:48.099+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:30:48 WARNING mlflow.utils.git_utils: Failed to import Git (the Git executable is probably not on your PATH), so Git SHA is not available. Error: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh(<full-path-to-git-executable>)

All git commands will error until this is rectified.

This initial message can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|silent|none|n|0: for no message or exception
    - warn|w|warning|log|l|1: for a warning message (logging level CRITICAL, displayed by default)
    - error|e|exception|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet
[2025-10-05T08:30:48.160+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/joblib/parallel.py:1379: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = self._backend.configure(

[2025-10-05T08:30:48.161+0000] {logging_mixin.py:190} INFO - Fitting 3 folds for each of 12 candidates, totalling 36 fits
[2025-10-05T08:31:39.764+0000] {logging_mixin.py:190} INFO -   Mejores par√°metros: {'classifier__max_depth': 20, 'classifier__min_samples_split': 2, 'classifier__n_estimators': 100}
[2025-10-05T08:31:45.871+0000] {logging_mixin.py:190} INFO -   Train Accuracy: 0.9999
[2025-10-05T08:31:45.871+0000] {logging_mixin.py:190} INFO -   Test Accuracy: 0.9737
[2025-10-05T08:31:45.871+0000] {logging_mixin.py:190} INFO -   Test F1 (weighted): 0.9736
[2025-10-05T08:31:45.871+0000] {logging_mixin.py:190} INFO -   CV Score: 0.9595 (+/- 0.0030)
[2025-10-05T08:31:48.125+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:31:48 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-10-05T08:31:48.175+0000] {credentials.py:1147} INFO - Found credentials in environment variables.
[2025-10-05T08:31:48.432+0000] {logging_mixin.py:190} WARNING - Registered model 'forest_cover_randomforest' already exists. Creating a new version of this model...
[2025-10-05T08:31:48.451+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:31:48 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: forest_cover_randomforest, version 7
[2025-10-05T08:31:48.451+0000] {logging_mixin.py:190} WARNING - Created version '7' of model 'forest_cover_randomforest'.
[2025-10-05T08:31:48.473+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:31:48 INFO mlflow.tracking._tracking_service.client: üèÉ View run RandomForest_batch_6 at: http://mlflow:5000/#/experiments/1/runs/ae99fcb6207c414da489d3baedcab9a8.
[2025-10-05T08:31:48.474+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:31:48 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow:5000/#/experiments/1.
[2025-10-05T08:31:48.493+0000] {logging_mixin.py:190} INFO - 
==================================================
[2025-10-05T08:31:48.494+0000] {logging_mixin.py:190} INFO - Modelo: GradientBoosting
[2025-10-05T08:31:48.494+0000] {logging_mixin.py:190} INFO - ==================================================
[2025-10-05T08:31:48.607+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/joblib/parallel.py:1379: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = self._backend.configure(

[2025-10-05T08:31:48.607+0000] {logging_mixin.py:190} INFO - Fitting 3 folds for each of 8 candidates, totalling 24 fits
[2025-10-05T08:39:01.797+0000] {logging_mixin.py:190} INFO -   Mejores par√°metros: {'classifier__learning_rate': 0.1, 'classifier__max_depth': 5, 'classifier__n_estimators': 100}
[2025-10-05T08:40:25.047+0000] {logging_mixin.py:190} INFO -   Train Accuracy: 0.9744
[2025-10-05T08:40:25.048+0000] {logging_mixin.py:190} INFO -   Test Accuracy: 0.9505
[2025-10-05T08:40:25.048+0000] {logging_mixin.py:190} INFO -   Test F1 (weighted): 0.9507
[2025-10-05T08:40:25.048+0000] {logging_mixin.py:190} INFO -   CV Score: 0.9407 (+/- 0.0048)
[2025-10-05T08:40:26.591+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:40:26 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-10-05T08:40:26.655+0000] {logging_mixin.py:190} WARNING - Registered model 'forest_cover_gradientboosting' already exists. Creating a new version of this model...
[2025-10-05T08:40:26.670+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:40:26 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: forest_cover_gradientboosting, version 7
[2025-10-05T08:40:26.671+0000] {logging_mixin.py:190} WARNING - Created version '7' of model 'forest_cover_gradientboosting'.
[2025-10-05T08:40:26.678+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:40:26 INFO mlflow.tracking._tracking_service.client: üèÉ View run GradientBoosting_batch_6 at: http://mlflow:5000/#/experiments/1/runs/5b017505cdea4c4894c2485568763942.
[2025-10-05T08:40:26.678+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:40:26 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow:5000/#/experiments/1.
[2025-10-05T08:40:26.687+0000] {logging_mixin.py:190} INFO - 
==================================================
[2025-10-05T08:40:26.687+0000] {logging_mixin.py:190} INFO - Modelo: LogisticRegression
[2025-10-05T08:40:26.687+0000] {logging_mixin.py:190} INFO - ==================================================
[2025-10-05T08:40:26.747+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/joblib/parallel.py:1379: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = self._backend.configure(

[2025-10-05T08:40:26.748+0000] {logging_mixin.py:190} INFO - Fitting 3 folds for each of 6 candidates, totalling 18 fits
[2025-10-05T08:41:04.951+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/sklearn/linear_model/_sag.py:349: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(

[2025-10-05T08:41:10.498+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/sklearn/linear_model/_sag.py:349: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(

[2025-10-05T08:41:16.184+0000] {warnings.py:110} WARNING - /home/airflow/.local/lib/python3.11/site-packages/sklearn/linear_model/_sag.py:349: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(

[2025-10-05T08:41:22.199+0000] {logging_mixin.py:190} INFO -   Mejores par√°metros: {'classifier__C': 10.0, 'classifier__solver': 'lbfgs'}
[2025-10-05T08:41:36.780+0000] {logging_mixin.py:190} INFO -   Train Accuracy: 0.7652
[2025-10-05T08:41:36.780+0000] {logging_mixin.py:190} INFO -   Test Accuracy: 0.7668
[2025-10-05T08:41:36.780+0000] {logging_mixin.py:190} INFO -   Test F1 (weighted): 0.7569
[2025-10-05T08:41:36.781+0000] {logging_mixin.py:190} INFO -   CV Score: 0.7641 (+/- 0.0042)
[2025-10-05T08:41:38.388+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:41:38 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-10-05T08:41:38.460+0000] {logging_mixin.py:190} WARNING - Registered model 'forest_cover_logisticregression' already exists. Creating a new version of this model...
[2025-10-05T08:41:38.501+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:41:38 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: forest_cover_logisticregression, version 7
[2025-10-05T08:41:38.501+0000] {logging_mixin.py:190} WARNING - Created version '7' of model 'forest_cover_logisticregression'.
[2025-10-05T08:41:38.514+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:41:38 INFO mlflow.tracking._tracking_service.client: üèÉ View run LogisticRegression_batch_6 at: http://mlflow:5000/#/experiments/1/runs/a3b81168ab6c4d9b9c0e189ce6db4abe.
[2025-10-05T08:41:38.514+0000] {logging_mixin.py:190} WARNING - 2025/10/05 08:41:38 INFO mlflow.tracking._tracking_service.client: üß™ View experiment at: http://mlflow:5000/#/experiments/1.
[2025-10-05T08:41:38.525+0000] {logging_mixin.py:190} INFO - 
==================================================
[2025-10-05T08:41:38.525+0000] {logging_mixin.py:190} INFO - ‚úì Entrenamiento completado
[2025-10-05T08:41:38.526+0000] {logging_mixin.py:190} INFO -   Mejor modelo: RandomForest (Accuracy: 0.9737)
[2025-10-05T08:41:38.526+0000] {logging_mixin.py:190} INFO - ==================================================
[2025-10-05T08:41:38.526+0000] {python.py:240} INFO - Done. Returned value was: {'results': {'RandomForest': {'test_accuracy': 0.9736906810917138, 'test_f1_weighted': 0.9736124343596074, 'cv_mean_score': 0.9594602116598868, 'best_params': {'classifier__max_depth': 20, 'classifier__min_samples_split': 2, 'classifier__n_estimators': 100}}, 'GradientBoosting': {'test_accuracy': 0.9504548807474797, 'test_f1_weighted': 0.9507196675540107, 'cv_mean_score': 0.9406809121234311, 'best_params': {'classifier__learning_rate': 0.1, 'classifier__max_depth': 5, 'classifier__n_estimators': 100}}, 'LogisticRegression': {'test_accuracy': 0.7667814113597247, 'test_f1_weighted': 0.7568528051465698, 'cv_mean_score': 0.7641380064153666, 'best_params': {'classifier__C': 10.0, 'classifier__solver': 'lbfgs'}}}, 'best_model': 'RandomForest', 'best_score': 0.9736906810917138, 'experiment_name': 'forest_cover_classification', 'mlflow_uri': 'http://mlflow:5000'}
[2025-10-05T08:41:38.554+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-10-05T08:41:38.555+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=forest_cover_mlops_pipeline, task_id=train_and_log_models, run_id=scheduled__2025-10-05T08:25:00+00:00, execution_date=20251005T082500, start_date=20251005T083047, end_date=20251005T084138
[2025-10-05T08:41:38.614+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-10-05T08:41:38.629+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-10-05T08:41:38.630+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
